HotelManagement/src/core/Room.java

```diff
@@ -34,25 +47,198 @@ private void setNumber(int number) {
+	public boolean isBooked(LocalDate fromDate, LocalDate toDate, int size) {
+		ArrayList<String> availableDates = findAvailableDatesForIntervalAndSize(fromDate, toDate, size);
+		String date = fromDate.getYear() + "-" + fromDate.getMonthValue() + "-" + fromDate.getDayOfMonth() + " to " +
+			toDate.getYear() + "-" + toDate.getMonthValue() + "-" + toDate.getDayOfMonth();
+		if (availableDates == null) {
+			return false;
+		}
+
+		return true;
 	}
 
-	/**
-	 * Books the room.
-	 */
-	public void bookTheRoom() {
-		this.isBooked = true;
+	public Set<AbstractCommodity> getCommodities() {
+		return commodities;
 	}
 
-	/**
-	 * Clear booking of the room.
-	 */
-	public void clearRoomBooking() {
-		this.isBooked = false;
+	public void prepareRoomCommodities() {
+		for(AbstractCommodity commodity : commodities) {
+			commodity.prepare();
+		}
+
+		maintenanceDates.add(LocalDateTime.now());
+	}
+
+	public int getNumberOfBeds() {
+		int numberOfBeds = 0;
+		for (AbstractCommodity commodity : commodities) {
+			if (commodity instanceof Bed) {
+				numberOfBeds++;
+			}
+		}
+
+		return numberOfBeds;
+	}
+
+	public void createBooking(LocalDate fromDate, LocalDate toDate, int size, String guestName, String guestId) {
+		ArrayList<String> availableDates = findAvailableDatesForIntervalAndSize(fromDate, toDate, size);
+		String date = fromDate.getYear() + "-" + fromDate.getMonthValue() + "-" + fromDate.getDayOfMonth() + " to " +
+			toDate.getYear() + "-" + toDate.getMonthValue() + "-" + toDate.getDayOfMonth();
+		if (availableDates == null) {
+			System.out.println("There are not available dates for this interval and size.");
+		} else if (availableDates.contains(date)) {
+			Booking booking = new Booking(fromDate, toDate, guestName, guestId);
+			bookings.add(booking);
+		} else {
+			System.out.println("Available dates in this interval:");
+			for (String availableDate : availableDates) {
+				System.out.println(availableDate);
+			}
+		}
+	}
+
+	public void removeBooking(LocalDate fromDate, LocalDate toDate) {
+		for (Booking booking : bookings) {
+			if (booking.getFromDate().isEqual(fromDate) && booking.getToDate().isEqual(toDate)) {
+				bookings.remove(booking);
+			}
+		}
+	}
+
+	private ArrayList<String> findBookedDays(LocalDate fromDate, LocalDate toDate) {
+		ArrayList<String> bookedDays = new ArrayList<>();
+		for (Booking booking : bookings) {
+			if (booking.getFromDate().getMonthValue() == fromDate.getMonthValue() ||
+				booking.getToDate().getMonthValue() == fromDate.getMonthValue() ||
+				booking.getFromDate().getMonthValue() == toDate.getMonthValue() ||
+				booking.getToDate().getMonthValue() == toDate.getMonthValue()) {
+				int daysInMonthOfBookingFromDate = booking.getFromDate().lengthOfMonth();
+				int day = booking.getFromDate().getDayOfMonth();
+				int month = booking.getFromDate().getMonthValue();
+				while (true) {
+					if (month == booking.getToDate().getMonthValue() && day == booking.getToDate().getDayOfMonth()) {
```

Comment:
If there happens to be a case where this if statement doesn't become true this method is a infinite loop hazard.

Reply:
Done